wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/fs/server/impl/ZipArchiver.java

```diff
@@ -111,6 +111,7 @@ void unzip(
             }
           }
 
+          Files.createDirectories(path.getParent());
```

Comment:
Does this code respect `withParents` param?

Reply:
What is "withParents" supposed to do? The current code fails if the first entry in the zip file is _not_ an entry directly inside the root directory. Unfortunately, I haven't found any documentation on "withParents".

Reply:
I dare to assume that the answer is "no".

From the source code I see that `withParents` parameter should indicates if there is a need to create not existing parent directories in `fsPath`. However if I understand correctly, your solution ignores this parameter and always creates missing directories. (I can not immediately say in what cases this parameter is useful, but we are fixing the bug and not discussing the API here.) Thus I consider that provided solution breaks the API.

I have to admit that a small part of che code is not well documented so next time if you will have difficulties with understanding the source code or the API please do not hesitate to ask che community members (via mailing list or mattermost).